### ISSUE TITLE
15$: Report warning on expression after newline after `return`

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#207]
Examples:

```js
function f() {
  let x = 42;
  return   // warning
    x;
}

function g() {
  let x = 42;
  if (x == 0) return  // no warning
  x;
}
```

**Comment by strager** [comment]
For the first case, if the token following `return` has `has_leading_newline` and is an identifier, emit the warning. Ideally, we'd emit the warning only if there's a difference in indentation, but we don't track indentation currently.

For the second case, we can add a flag `parse_and_visit_statement`. (I think the `allow_declarations` flag might be exactly what we want, inverted.) If the flag says we're in a curly-less `if`/`while`/whatever, don't emit the warning.

**Comment by strager** [comment]
This is working. E0179.
